modules/dnn/test/test_tf_importer.cpp

```diff
@@ -568,6 +569,41 @@ TEST_P(Test_TensorFlow_layers, l2_normalize_3d)
     runTensorFlowNet("l2_normalize_3d");
 }
 
+class Test_TensorFlow_diagnostics : public DNNTestLayer {
+public:
+    Test_TensorFlow_diagnostics() : logLevel(cv::utils::logging::getLogLevel())
+    {
+        enableModelDiagnostics(true);
+    }
+
+    ~Test_TensorFlow_diagnostics()
+    {
+        enableModelDiagnostics(false);
+    }
+
+    void runFailingTensorFlowNet(const std::string& prefix, bool hasText = false)
+    {
+        std::string netPath = path(prefix + "_net.pb");
+        std::string netConfig = (hasText ? path(prefix + "_net.pbtxt") : "");
+
+        Net net = readNetFromTensorflow(netPath, netConfig);
+    }
+private:
+    cv::utils::logging::LogLevel logLevel;
```

Comment:
is it used?
should we restore log level in destructor?

Reply:
I deleted it already. Originally, yes, in constructor I was silencing logs and in the destructor the appropriate level was restored, but in event of test failure we would loose logs. Report.xml shouldn't contain stderr output and I decided to leave logs be.